### PR TITLE
[OPTEE-OS 3/5] pkg: Add new package to build OPTEE-OS

### PR DESCRIPTION
**Pull request by rene**
Add a new package that builds OPTEE-OS for QEMU and i.MX8MP platforms.

:warning: This PR depends on the following PRs:

- https://github.com/lf-edge/eve/pull/3126
- https://github.com/lf-edge/eve/pull/3127

It should be merged only after all dependencies are merged.

### CODE REVIEW

**Comment by rene** [comment]
Remove TPM driver/event configuration since they are not needed (will be needed only when we integrate fTPM PTA).

**Comment by rene** [comment]
Updates in this PR:

Update eve-alpine version tag.

**Review by rouming — APPROVED** [review]
Ack

File: pkg/optee-os/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ENV PKGS musl-dev libgcc libintl libuuid libtirpc libblkid libattr
+RUN eve-alpine-deploy.sh
+
+# adjust EVE_TARGET_ARCH for cross-compiler
+FROM build-cross AS build-cross-target-amd64
+ENV EVE_TARGET_ARCH=x86_64
+FROM build-cross AS build-cross-target-arm64
+ENV EVE_TARGET_ARCH=aarch64
+
+# hadolint ignore=DL3006
```
Comment (christoph-zededa) [inline pkg/optee-os/Dockerfile:34]: I think at some point in the future we should ignore DL3006 in general ...

File: pkg/optee-os/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            CFG_TEE_CORE_LOG_LEVEL=1 \
+            PLATFORM="${target}" \
+            PLATFORM_FLAVOR="${flavor}" \
+            CFG_UART_BASE="${uartbase}" \
+            O=out/arm \
+            ${customdef} ;\
+         mkdir -p "/opteeos_bins/${t}" ;\
+         cp out/arm/core/tee-raw.bin "/opteeos_bins/${t}/optee.bin" ;\
+         cp -r out/arm/export-ta_arm64 "/opteeos_bins/${t}/export-ta_arm64") ;\
+    done
```
Comment (christoph-zededa) [inline pkg/optee-os/Dockerfile:127]: Isn't it possible to split it into several `RUN`s?

Comment (eriknordmark) [reply pkg/optee-os/Dockerfile:127]: @christoph-zededa there used to be q yetus complaints when there was a sequence of RUN statements, and I don't know if we turned off that complaint.

**Review by eriknordmark — APPROVED** [review]
Run eden

File: pkg/optee-os/Dockerfile (hunk 2 above)
Comment (christoph-zededa) [reply pkg/optee-os/Dockerfile:127]: We turned it off: https://github.com/lf-edge/eve/blob/master/.hadolint.yaml

**Pull request merged**